.github/workflows/tests.yaml

```diff
@@ -133,13 +133,26 @@ jobs:
           clang++ --version
 
       # Extract our access key for our build cache.
-      - name: Extract access key
-        if: steps.filter.outputs.ignore == 'false'
+      - name: Extract access key (read-only)
+        if:
+          steps.filter.outputs.ignore == 'false' && github.event_name ==
+          'pull_request_target'
```

Comment:
Just to confirm -- the goal is to first split the keys, and then pivot to just `pull_request` actions (and make they key visible there)?

Reply:
No, this is the final state.

If you're referring to my comment, pull requests use the merged, trunk state of the action, which I can't modify with a pending PR. So I can test the results of a push event (because I can push to action-test) but not a pull request event (because I cannot push to trunk).

Reply:
My understanding is that we'll need to move from `pull_request_target` to `pull_request` to fully secure this: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

I still think this is a good first step though? Happy to chat more about this if useful.

Reply:
Switched. Note, still only moderately tested.